tests/test_abort.js

```diff
@@ -1,157 +1,180 @@
 'use strict'
 
-const nock = require('..')
 const { expect } = require('chai')
 const http = require('http')
 const sinon = require('sinon')
+const nock = require('..')
 
 require('./setup')
 
-it('`req.abort()` should cause "abort" and "error" to be emitted', done => {
-  const scope = nock('http://example.test')
-    .get('/')
-    .delayConnection(500)
-    .reply()
-
-  const onAbort = sinon.spy()
-  const req = http
-    .get('http://example.test/')
-    .once('abort', onAbort)
-    .once('error', err => {
-      // Should trigger last
-      expect(err.code).to.equal('ECONNRESET')
-      expect(onAbort).to.have.been.calledOnce()
-      scope.done()
-      done()
-    })
-  process.nextTick(() => req.abort())
-})
+// These tests use `setTimeout` before verifying emitted events to ensure any
+// number of `nextTicks` or `setImmediate` can process first.
 
-it('abort is emitted before delay time', done => {
-  const scope = nock('http://example.test')
-    .get('/')
-    .delayConnection(500)
-    .reply()
-
-  const start = Date.now()
-  const req = http
-    .get('http://example.test/')
-    .once('abort', () => {
-      const actual = Date.now() - start
-      expect(actual).to.be.below(250)
-      scope.done()
+// Node will emit a `prefinish` event after `socket`, but it's an internal,
+// undocumented event that Nock does not emulate.
+
+// The order of tests run sequentially through a ClientRequest's lifetime.
+// Starting the top by aborting requests early on then aborting later and later.
+describe('`ClientRequest.abort()`', () => {
+  it('should not emit an error when `write` is called on an aborted request', done => {
```

Comment:
I wonder if all of these tests should say e.g. "Emits the expected event sequence when `write` is called on an aborted request".